src/client/components/fields/DateField.tsx

```diff
@@ -8,46 +8,64 @@ import {
   FormHelperText,
   FormErrorMessage,
   Input,
+  InputGroup,
+  InputRightElement,
 } from '@chakra-ui/react'
 
 import 'react-datepicker/dist/react-datepicker.css'
 import '../../styles/date-picker.css'
 
 import { Field } from '../../../types/checker'
+import { BiCalendar } from 'react-icons/bi'
+
+interface DatePickerInputProps {
+  value: string
+  onClick: (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => void
+}
+
+const DatePickerInput: FC<DatePickerInputProps> = ({ value, onClick }) => (
+  <InputGroup>
+    <Input
+      type="text"
+      placeholder="DD/MM/YYYY"
+      className="react-datepicker-ignore-onclickoutside"
+      value={value}
+      onClick={onClick}
+    />
+    <InputRightElement pointerEvents="none" children={<BiCalendar />} />
+  </InputGroup>
+)
 
 export const DateField: FC<Field> = ({ id, title, description }) => {
   const styles = useStyles()
-  const { register, errors } = useFormContext()
-  const [date, setDate] = useState(new Date())
+  const { control, errors } = useFormContext()
   const error = errors[id]
 
-  const changeHandler = (newDate: Date) => {
-    if (newDate) setDate(newDate)
-  }
-
   return (
-    <FormControl isInvalid={error} id={id}>
-      <FormLabel sx={styles.label} htmlFor={id}>
-        {title}
-      </FormLabel>
-      {description && <FormHelperText mb={4}>{description}</FormHelperText>}
-      <Input
-        type="hidden"
-        name={id}
-        value={date ? date.valueOf() : undefined}
-        ref={register({ required: true })}
-      />
-      <DatePicker
-        wrapperClassName={error ? 'fieldError' : ''}
-        value={date ? date.toISOString().split('T')[0] : ''}
-        selected={date}
-        onChange={changeHandler}
-        showPopperArrow={true}
-        showMonthDropdown
-        showYearDropdown
-        dropdownMode="select"
-      />
-      <FormErrorMessage>Field is required</FormErrorMessage>
-    </FormControl>
+    <Controller
```

Comment:
Validation for the `DateField` doesn't seem to be triggering. Error is only thrown at run time. 

![image](https://user-images.githubusercontent.com/3666479/109468669-e4b31200-7aa7-11eb-98bb-c527dcb13c17.png)